src/controllers/ping.js

```diff
@@ -1,25 +1,28 @@
 import {
   dbGetMonitorByEndpointKey,
-  dbMonitorRecovery,
+  dbUpdateMonitorRecovered,
   dbUpdateNextAlert,
   dbAddPing,
 } from '../db/queries.js';
 
-const addPing = async (req, res) => {
+const addPing = async (req, res, next) => {
   try {
     const endpoint_key = req.params.endpoint_key;
     const monitor = await dbGetMonitorByEndpointKey(endpoint_key);
+    if (!monitor) {
+      throw new Error('Unable to find monitor associated with that endpoint.');
+    }
 
-    dbAddPing(monitor.id);
+    await dbAddPing(monitor.id); // need return value?
 
     if (monitor.failing) {
-      dbMonitorRecovery(monitor.id);
+      await dbUpdateMonitorRecovered(monitor.id); // need return value?
     }
 
-    dbUpdateNextAlert(endpoint_key);
+    await dbUpdateNextAlert(monitor); // need return value?
```

Comment:
No we don't need the return values here which is why I originally called without await - you're right about the error handling not working without them though.